### PR TITLE
chore(dockerfile): bump to JDK 17

### PR DESCRIPTION
**Pull request by yanavasileva**
https://github.com/camunda/camunda-bpm-platform/issues/3434

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -144,11 +143,12 @@ disables Swagger UI by default.
 Our docker images are using the latest LTS OpenJDK version supported by
 Camunda Platform. This currently means:
 
- - Camunda 7.12 or later will be based on OpenJDK 11.
+ - Camunda 7.20 or later will be based on OpenJDK 17.
+ - Camunda 7.12 - 7.19 is be based on OpenJDK 11.
```
Comment (danielkelemen) [inline README.md]: 🔧 

```suggestion
 - Camunda 7.12 - 7.19 is based on OpenJDK 11.
```

File: README.md [hunk 2]
```diff
@@ -144,11 +143,12 @@ disables Swagger UI by default.
...
  - All previous versions are based on OpenJDK 8.
 
-While all the OpenJDK versions supported by Camunda will work, we will not
-provide a ready to use image for them.
+While all the OpenJDK versions supported by Camunda will work for Tomcat and WildFly, we will not
+provide a ready to use image for them. (Camunda Run supports only JDK 17.)
```
Comment (danielkelemen) [inline README.md]: 💬 Wildfly also needs a minimum Java 11. I'm not sure how to formulate this.

**Review by danielkelemen — APPROVED** [review]
Looks good, left 2 minor notes.

File: README.md (hunk 2 above)
Comment (yanavasileva) [reply README.md]: 👍 True, I will make an adjustment.

**Comment by yanavasileva** [comment]
@danielkelemen could you have a look at the readme again?

File: README.md [hunk 3]
```diff
@@ -144,11 +143,14 @@ disables Swagger UI by default.
 Our docker images are using the latest LTS OpenJDK version supported by
 Camunda Platform. This currently means:
 
- - Camunda 7.12 or later will be based on OpenJDK 11.
+ - Camunda 7.20 or later will be based on OpenJDK 17.
+   - Camunda 7.19 image for Camunda Run is supported only for JDK 17.
```
Comment (danielkelemen) [inline README.md]: ❓ shouldn't this be 7.20 for Run?

```suggestion
   - Camunda 7.20 image for Camunda Run is supported only for JDK 17.
```

File: README.md [hunk 4]
```diff
@@ -144,11 +143,14 @@ disables Swagger UI by default.
...
+ - Camunda 7.12 - 7.19 is based on OpenJDK 11.
+   - Camunda 7.19 image for WildFly is supported only for JDK 11 and JDK 17.
  - All previous versions are based on OpenJDK 8.
 
-While all the OpenJDK versions supported by Camunda will work, we will not
-provide a ready to use image for them.
+While all the OpenJDK versions supported by Camunda will work (without the exceptions specified above),
+we will not provide a ready to use image for them.
```
Comment (danielkelemen) [inline README.md]: 🔧 Maybe sounds a bit better.

```suggestion
While all the OpenJDK versions supported by Camunda will work with the exceptions specified above,
we will not provide ready to use images for them.
```

**Review by danielkelemen — APPROVED** [review]
Sounds good. I left 2 minor notes again :)

**Comment by yanavasileva** [comment]
I will wait merging the changes until issues on GHA for 7.20 are resolved.

**Pull request merged**